Unable to build SpringContext with MicronautApplicationContext as parent

### Expected Behavior

Run an application with SpringContext and MicronautApplicationContext as parent.

### Actual Behaviour

Application won't start.

Same was working with micronaut 3.

<details>
<summary>Error</summary>
<pre>
Exception in thread "main" io.micronaut.core.io.service.SoftServiceLoader$ServiceLoadingException: Failed to load a service: io.micronaut.context.exceptions.BeanContextException: Unexpected error loading bean definition [com.example.$Application$AutoConfigurationPackages$Registrar0$Definition]: failed to access class org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar from class com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference (org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar and com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference are in unnamed module of loader 'app')
	at io.micronaut.core.io.service.ServiceScanner$ServiceInstanceLoader.collect(ServiceScanner.java:364)
	at io.micronaut.core.io.service.ServiceScanner$MicronautMetaServicesLoader.collect(ServiceScanner.java:289)
	at io.micronaut.core.io.service.ServiceScanner$DefaultServiceCollector.collect(ServiceScanner.java:246)
	at io.micronaut.core.io.service.SoftServiceLoader.collectDynamicServices(SoftServiceLoader.java:201)
	at io.micronaut.core.io.service.SoftServiceLoader.collectAll(SoftServiceLoader.java:176)
	at io.micronaut.context.DefaultBeanContext.resolveBeanDefinitionReferences(DefaultBeanContext.java:1865)
	at io.micronaut.context.DefaultApplicationContext.isBootstrapPropertySourceLocatorPresent(DefaultApplicationContext.java:162)
	at java.base/java.util.Objects.requireNonNullElseGet(Objects.java:310)
	at io.micronaut.context.DefaultApplicationContext.isBootstrapEnabled(DefaultApplicationContext.java:158)
	at io.micronaut.context.DefaultApplicationContext.createEnvironment(DefaultApplicationContext.java:146)
	at io.micronaut.context.DefaultApplicationContext.getEnvironment(DefaultApplicationContext.java:180)
	at io.micronaut.context.DefaultApplicationContextBuilder.build(DefaultApplicationContextBuilder.java:322)
	at io.micronaut.spring.context.MicronautApplicationContext.<init>(MicronautApplicationContext.java:114)
	at io.micronaut.spring.context.MicronautApplicationContext.<init>(MicronautApplicationContext.java:106)
	at com.example.Application.main(Application.groovy:16)
Caused by: io.micronaut.core.io.service.SoftServiceLoader$ServiceLoadingException: io.micronaut.context.exceptions.BeanContextException: Unexpected error loading bean definition [com.example.$Application$AutoConfigurationPackages$Registrar0$Definition]: failed to access class org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar from class com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference (org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar and com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference are in unnamed module of loader 'app')
	at io.micronaut.core.io.service.SoftServiceLoader.lambda$collectDynamicServices$2(SoftServiceLoader.java:197)
	at io.micronaut.core.io.service.ServiceScanner$ServiceInstanceLoader.compute(ServiceScanner.java:355)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: io.micronaut.context.exceptions.BeanContextException: Unexpected error loading bean definition [com.example.$Application$AutoConfigurationPackages$Registrar0$Definition]: failed to access class org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar from class com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference (org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar and com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference are in unnamed module of loader 'app')
	at io.micronaut.context.AbstractInitializableBeanDefinitionReference.isPresent(AbstractInitializableBeanDefinitionReference.java:205)
	at io.micronaut.core.io.service.SoftServiceLoader.lambda$collectDynamicServices$2(SoftServiceLoader.java:190)
	... 7 more
Caused by: java.lang.IllegalAccessError: failed to access class org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar from class com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference (org.springframework.boot.autoconfigure.AutoConfigurationPackages$Registrar and com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference are in unnamed module of loader 'app')
	at com.example.$Application$AutoConfigurationPackages$Registrar0$Definition$Reference.getBeanType(Unknown Source)
	at io.micronaut.context.AbstractInitializableBeanDefinitionReference.isPresent(AbstractInitializableBeanDefinitionReference.java:197)
	... 8 more
</pre>
</details>



### Steps To Reproduce

1. create a micronaut application
2. add spring
3. Replace main in Application.class with:
  ```groovy
  new SpringApplicationBuilder()
                .parent(new MicronautApplicationContext().tap { start() })
                .sources(Application)
                .build()
                .run()
  ```

### Environment Information

OS: Linux
JDK: 17 & 21

### Example Application

https://github.com/safri-net/mn4-spring-issue

### Version

4.2.3

--- a/spring-annotation/src/main/java/io/micronaut/spring/annotation/beans/ImportAnnotationVisitor.java
+++ b/spring-annotation/src/main/java/io/micronaut/spring/annotation/beans/ImportAnnotationVisitor.java
@@ -74,17 +74,21 @@ public final class ImportAnnotationVisitor implements TypeElementVisitor<Object,
         if (annType != null && element.hasStereotype(annType)) {
             List<AnnotationValue<? extends Annotation>> values = element.getAnnotationValuesByType(annType);
             if (!values.isEmpty()) {
-                for (AnnotationValue<?> av : values) {
-                    AnnotationClassValue<?>[] acv = av.annotationClassValues(AnnotationMetadata.VALUE_MEMBER);
-                    for (AnnotationClassValue<?> a : acv) {
-                        String className = a.getName();
-                        context.getClassElement(className).ifPresent(typeToImport -> {
-                            if (typeToImport.isPublic()) {
-                                handleImport(element, context, typeToImport);
-                            }
-                        });
+                visitValues(element, context, values);
+            }
+        }
+    }
+
+    private void visitValues(ClassElement element, VisitorContext context, List<AnnotationValue<? extends Annotation>> values) {
+        for (AnnotationValue<?> av : values) {
+            AnnotationClassValue<?>[] acv = av.annotationClassValues(AnnotationMetadata.VALUE_MEMBER);
+            for (AnnotationClassValue<?> a : acv) {
+                String className = a.getName();
+                context.getClassElement(className).ifPresent(typeToImport -> {
+                    if (typeToImport.isPublic()) {
+                        handleImport(element, context, typeToImport);
                     }
-                }
+                });
             }
         }
     }

--- a/spring-annotation/src/main/java/io/micronaut/spring/annotation/beans/ImportAnnotationVisitor.java
+++ b/spring-annotation/src/main/java/io/micronaut/spring/annotation/beans/ImportAnnotationVisitor.java
@@ -58,8 +58,7 @@ import io.micronaut.spring.beans.ImportedBy;
 import io.micronaut.spring.core.type.ClassElementSpringMetadata;
 
 /**
- * Handles the import importDeclaration allowing importing of additional Spring beans into a Micronaut
- application.
+ * Handles the import importDeclaration allowing importing of additional Spring beans into a Micronaut application.
  *
  * @author graemerocher
  * @since 4.3.0
@@ -80,7 +79,9 @@ public final class ImportAnnotationVisitor implements TypeElementVisitor<Object,
                     for (AnnotationClassValue<?> a : acv) {
                         String className = a.getName();
                         context.getClassElement(className).ifPresent(typeToImport -> {
-                            handleImport(element, context, typeToImport);
+                            if (typeToImport.isPublic()) {
+                                handleImport(element, context, typeToImport);
+                            }
                         });
                     }
                 }

